### ISSUE TITLE
2d L intersection

### ISSUE THREAD

**Issue by alessiofumagalli** [pmgbergen/porepy#22]
The L intersections are not working in 2d. Minimal example:
```python
from porepy.viz import plot_grid
from porepy.fracs import importer

mesh_kwargs = {}
mesh_kwargs['mesh_size'] = {'mode': 'constant',
                            'value': 0.5, 'bound_value': 0.5}

domain = {'xmin': 0, 'xmax': 1, 'ymin': 0, 'ymax': 1}
gb = importer.from_csv('single.csv', mesh_kwargs, domain)
gb.compute_geometry()
plot_grid.plot_grid(gb, info="all", alpha=0)
```
where single.csv is
```txt
FID,START_X,START_Y,END_X,END_Y
1,0.2,0.5,0.8,0.5
2,0.2,0.1,0.2,0.5
```
I get the following error
```shell
    gb = importer.from_csv('single.csv', mesh_kwargs, domain)
  File "/home/elle/ANIGMA/porepy/src/porepy/fracs/importer.py", line 56, in from_csv
    return meshing.simplex_grid(f_set, domain, **mesh_kwargs)
  File "/home/elle/ANIGMA/porepy/src/porepy/fracs/meshing.py", line 72, in simplex_grid
    grids = simplex.triangle_grid(frac_dic, domain, **kwargs)
  File "/home/elle/ANIGMA/porepy/src/porepy/fracs/simplex.py", line 253, in triangle_grid
    pts, cells, phys_names, cell_info, line_tag=const.PHYSICAL_NAME_FRACTURES)
  File "/home/elle/ANIGMA/porepy/src/porepy/grids/gmsh/mesh_2_grid.py", line 157, in create_1d_grids
    g = create_embedded_line_grid(loc_coord, loc_pts_1d)
  File "/home/elle/ANIGMA/porepy/src/porepy/grids/gmsh/mesh_2_grid.py", line 186, in create_embedded_line_grid
    tangent = cg.compute_tangent(loc_coord)
  File "/home/elle/ANIGMA/porepy/src/porepy/utils/comp_geom.py", line 1560, in compute_tangent
    assert not np.allclose(tangent, np.zeros(3))
AssertionError

```

**Comment by keileg** [comment]
There is a unique filtering of the points missing somewhere. I'm on it.

**Comment by alessiofumagalli** [comment]
Sorry, but with the modification of this issue I get a problem.
Minimal example:
```python
mesh_kwargs = {}
mesh_kwargs['mesh_size'] = {'mode': 'constant', 'value': 2000, 'bound_value': 2000}
gb = importer.from_csv('network.csv', mesh_kwargs)
```
where the network is described by the file:
```txt
# FID,START_X,START_Y,END_X,END_Y
0.000000000000000000e+00,1.762209000000000003e+02,0.000000000000000000e+00,1.228545399999999972e+03,1.150981799999000032e+03
1.000000000000000000e+00,1.038469999999999942e+02,8.960951999999999771e+02,7.500931000000000495e+02,1.456616099999999960e+03
2.000000000000000000e+00,5.456289999999999907e+01,5.027385999989999732e+02,9.059012000000000171e+02,1.208550899998999967e+03
3.000000000000000000e+00,7.289112999999999829e+02,3.762620000000000005e+02,1.016928099999999972e+03,6.331676999999999680e+02
4.000000000000000000e+00,2.737386000000000195e+02,1.043450499999999920e+03,3.713433999999999742e+02,1.420304599999999937e+03
5.000000000000000000e+00,2.541017999999999972e+02,1.026418599999999969e+03,2.879995000000000118e+02,1.273460600000000113e+03
6.000000000000000000e+00,6.470498000000000047e+02,9.939470000009999922e+02,7.767241999999999962e+02,6.568016000000000076e+02
7.000000000000000000e+00,3.340038999999999874e+02,1.276135999999999967e+03,4.990416999999999916e+02,1.238867000000999951e+03
8.000000000000000000e+00,3.996560000000000059e+02,1.152664899998999999e+03,7.524040999999999713e+02,1.081292299999000079e+03
9.000000000000000000e+00,1.334153000000000020e+02,9.217412000000000489e+02,7.380425999999999931e+02,1.084198100001000057e+03
1.000000000000000000e+01,7.761567000000000007e+02,1.003585099999000022e+03,7.954130999999999858e+02,6.655288000000000466e+02
```
I get the following error:
```shell
    gb = importer.from_csv('/home/elle/ANIGMA/anigma/fracgen_topology/mucca/original_porepy.csv', mesh_kwargs)
  File "/home/elle/ANIGMA/porepy/src/porepy/fracs/importer.py", line 56, in from_csv
    return meshing.simplex_grid(f_set, domain, **mesh_kwargs)
  File "/home/elle/ANIGMA/porepy/src/porepy/fracs/meshing.py", line 72, in simplex_grid
    grids = simplex.triangle_grid(frac_dic, domain, **kwargs)
  File "/home/elle/ANIGMA/porepy/src/porepy/fracs/simplex.py", line 213, in triangle_grid
    lines_split[:2] = old_2_new[lines_split[:2]]
IndexError: index 26 is out of bounds for axis 1 with size 26
```

**Comment by keileg** [comment]
There was an issue with point arrays after splitting of intersecting fractures.

Can you try to pull the branch fix_issue_22 and check if it is robust? If so, I will merge.

**Comment by keileg** [comment]
@alessiofumagalli Are you happy? Can I close?

**Comment by alessiofumagalli** [comment]
Yes, I'm happy.. So far :)

**Comment by keileg** [comment]
Resolved by #29
